File: Assets/Coin Game/Script/Coin.js
import UnityEngine;
import System.Collections;

public var CoinCount : int = 0; //Total # of Coins in Scene
public var Coins : GameObject[];

//Runs at the beginning of game
function Start () {

    //Debug.Log("Object Created");

    Coins = GameObject.FindGameObjectsWithTag("Coin");

    CoinCount = Coins.length;

}

//Runs every frame of game
function Update () {

    Coins = GameObject.FindGameObjectsWithTag("Coin");
    CoinCount = Coins.length;
    Collection();

}

function Collection(){

    if(CoinCount <= 0){
        Debug.Log("You Won!!");
    }
}

File: Assets/Coin Game/Script/Coin.cs
using UnityEngine;
using System.Collections;

public class Coin : MonoBehaviour {

    public static int CoinCount = 0; //Total # of Coins in Scene
    public static GameObject[] Coins;

    void Start() {

        //Debug.Log("Object Created");
        Coins = GameObject.FindGameObjectsWithTag("Coin");
        CoinCount = Coins.Length;

    }

    void Update() {

        Coins = GameObject.FindGameObjectsWithTag("Coin");
        CoinCount = Coins.Length;
        Collection();
    }

    void Collection() {

        if(CoinCount <= 0) {
            Debug.Log("You Won!!");
        }
    }

}
